apps/dashboard/src/components/settings.tsx

```diff
@@ -128,6 +128,50 @@ export function SettingText(props: {
   );
 }
 
+export function SettingSelect<TValue extends string>(props: {
+  label: string | React.ReactNode,
+  hint?: string | React.ReactNode,
+  value?: TValue,
+  placeholder?: string,
+  disabled?: boolean,
+  onValueChange?: (value: TValue) => void | Promise<void>,
+  actions?: React.ReactNode,
+  children: React.ReactNode,
+}) {
+  const id = useId();
+  const [valueState, setValueState] = useState<TValue | undefined>(props.value);
+  const value = props.value ?? valueState;
+
+  const onValueChange = async (value: TValue) => {
+    setValueState(value);
+    await props.onValueChange?.(value);
+  };
+
+  return (
+    <div className="flex flex-col gap-2">
+      <div className="flex items-center">
+        <Label className='pr-2' htmlFor={id}>{props.label}</Label>
+        {props.actions}
+      </div>
+      <Select
+        value={value}
+        onValueChange={(value: TValue) => runAsynchronouslyWithAlert(onValueChange(value))}
```

Comment:
If a component's events can run asynchronously, we should in some way indicate on the component that something is processing in the meantime (eg. a loading indicator)